specs/ClearBrowsingData.md

```diff
@@ -0,0 +1,233 @@
+  COREWEBVIEW2_BROWSING_DATA_KIND_HTTP_CACHE = 0x18,
+
+  /// Specifies download history data. 
+  COREWEBVIEW2_BROWSING_DATA_KIND_DOWNLOAD_HISTORY = 0x19, 
+
+  /// Specifies general autofill form data. 
+  /// This excludes password forms and includes information like: 
+  /// names, street and email addresses, phone numbers, and arbitrary input. 
+  COREWEBVIEW2_BROWSING_DATA_KIND_GENERAL_AUTOFILL = 0x20, 
+
+  /// Specifies password autofill forms data. 
+  COREWEBVIEW2_BROWSING_DATA_KIND_PASSWORD_AUTOFILL = 0x21,
+
+  /// Specifies browsing history data. 
+  COREWEBVIEW2_BROWSING_DATA_KIND_BROWSING_HISTORY = 0x22,
+
+  /// Specifies bookmarks data.  
+  COREWEBVIEW2_BROWSING_DATA_KIND_BOOKMARKS = 0x23,
+
+  /// Specifies settings data.
+  COREWEBVIEW2_BROWSING_DATA_KIND_SETTINGS = 0x24,
+
+  /// Specifies content settings data.
+  COREWEBVIEW2_BROWSING_DATA_KIND_CONTENT_SETTINGS = 0x25,
+
+  /// Specifies local custom dictionary data. 
+  COREWEBVIEW2_BROWSING_DATA_KIND_LOCAL_CUSTOM_DICTIONARY = 0x26,
+
+  /// Specifies profile data that should be wiped to make it look like a new profile. 
+  /// This does not delete account-scoped data like passwords but will remove access
+  /// to account-scoped data by signing the user out. 
+  /// This browsing data kind if inclusive of COREWEBVIEW2_BROWSING_DATA_KIND_SITE,
+  /// COREWEBVIEW2_BROWSING_DATA_KIND_HTTP_CACHE, COREWEBVIEW2_BROWSING_DATA_KIND_DOWNLOAD_HISTORY,
+  /// COREWEBVIEW2_BROWSING_DATA_KIND_GENERAL_AUTOFILL, COREWEBVIEW2_BROWSING_DATA_KIND_PASSWORD_AUTOFILL,
+  /// COREWEBVIEW2_BROWSING_DATA_KIND_BROWSING_HISTORY, COREWEBVIEW2_BROWSING_DATA_KIND_CONTENT_SETTINGS,
+  /// COREWEBVIEW2_BROWSING_DATA_KIND_BOOKMARKS, COREWEBVIEW2_BROWSING_DATA_KIND_SETTINGS,
+  /// COREWEBVIEW2_BROWSING_DATA_KIND_LOCAL_CUSTOM_DICTIONARY.
+  COREWEBVIEW2_BROWSING_DATA_KIND_PROFILE = 0X27,
+} COREWEBVIEW2_BROWSING_DATA_KIND;
+
+[uuid(4ecfcb16-dd09-4464-9a71-fd8e2d3ac0a2), object, pointer_default(unique)]
+interface ICoreWebView2SEnvironment5 : ICoreWebView2Environment {
+  /// Clear browsing data based on a specific data type. The completed handler
+  /// will be invoked when the browsing data has been cleared and will indicate 
+  /// if the specified data was properly cleared. 
+  HRESULT ClearBrowsingData(
+    [in] COREWEBVIEW2_BROWSING_DATA_KIND dataKind,
+    [in] ICoreWebView2ClearBrowsingDataCompletedHandler *handler);
+}
+
+[uuid(c2b78e49-5bf5-4d38-a535-668a8a8a30d9), object, pointer_default(unique)]
+interface ICoreWebView2ClearBrowsingDataCompletedHandler : IUnknown {
+  /// Provide the completion status and result of the corresponding
+  /// asynchronous method.
+  HRESULT Invoke([in] HRESULT errorCode, [in] BOOL isSuccessful);
+}
+```
+### .NET, WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    public partial class CoreWebView2Environment
+    {
+        public async Task<bool> ClearBrowsingDataAsync(CoreWebView2BrowsingDataKind dataKind);
```

Comment:
Is there a scenario for clearing all browsing data? If so, we could add an overload with no parameters that just clears everything.

In COM we can't do overloads so the name would need to be different but it could be in COM
```
HRESULT ClearAllBrowsingData()
```

And in .NET/WinRT
```
public async Task<bool> ClearBrowsingDataAsync();
```

Reply:
There is a scenario for clearing all browsing data - there are a couple asks for this. The profile data kind matches the chrome browsing data type ALL_DATA_TYPES. This chrome browsing data type ors all the data types that are listed above the profile data type.